### PR TITLE
fix metadata for R.addIndex

### PR DESCRIPTION
**Pull request by davidchambers**
The missing type breaks our script for generating the HTML documentation.

### CODE REVIEW

**Comment by davidchambers** [comment]
More accurately, it broke old versions of the script. @CrossEye added a workaround in https://github.com/ramda/ramda.github.io/commit/795cbe836e295f6fe627f0c35f2269d5cc0660dd, but we should fix the root cause regardless.

File: src/addIndex.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ var curryN = require('./curryN');
  * @category List
  * @sig ((a ... -> b) ... -> [a] -> *) -> (a ..., Int, [a] -> b) ... -> [a] -> *)
  * @param {Function} fn A list iteration function that does not pass index/list to its callback
- * @return An altered list iteration function thats passes index/list to its callback
+ * @return {Function} An altered list iteration function thats passes index/list to its callback
```
Comment (buzzdecafe) [inline src/addIndex.js]: > function **thats** passes index/list 

typo

and "item, index, and list" would be clearer

File: src/addIndex.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ var curryN = require('./curryN');
  * @category List
  * @sig ((a ... -> b) ... -> [a] -> *) -> (a ..., Int, [a] -> b) ... -> [a] -> *)
```
Comment (buzzdecafe) [inline src/addIndex.js]: > (a ..., Int, [a] -> b)

i think we should start wrapping these for clarity. For instance, this is not some `a`'s, an `Int`, and a function from `[a] -> b`. clearer IMO would be:

`((a ..., Int, [a]) -> b)`

Comment (davidchambers) [inline src/addIndex.js]: I don't understand the `...` in `(a ..., Int, [a])`. Shouldn't it just be `(a, Int, [a])`?

Can you share your suggestion for the entire signature?

Comment (buzzdecafe) [inline src/addIndex.js]: i just copied what you had there

Comment (davidchambers) [inline src/addIndex.js]: I don't want to change a signature I don't understand. Let's leave this for a separate pull request. :)

Comment (buzzdecafe) [inline src/addIndex.js]: sounds good. i also don't understand the intended meaning of the ellipsis after `a` in the function sig.

Comment (CrossEye) [inline src/addIndex.js]: I would welcome a clearer version of this signature.  But the function has some inherent complexity in that it's supposed to handle various kinds of cases, including:

**Map:**

    (inVal -> outVal) -> list -> result
    ((inVal, idx, list) -> outVal) -> list -> result

**Reduce**

    ((accumVal, currVal) -> newVal) -> initVal -> list -> result
    ((accumVal, currVal, idx, list) -> newVal) -> initVal -> list -> result

The nested ellipses are meant to represent the difference between `inVal -> outVal` and `(accumVal, currVal) -> newVal`, allowing for the possibility that a function might have still more parameters than this.  The top-level ellipses are to represent the difference between `fn -> list -> result` and `fn -> initVal -> list -> result`, again with the possibility that there might be even more parameters.

Any suggestions would be welcome.
